src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -125,10 +125,14 @@ int VpiArrayObjHdl::initialise(const std::string &name,
 
     m_indexable = true;
 
-    int range_idx = 0;
+    vpiHandle iter = vpi_iterate(vpiRange, hdl);
+
+    s_vpi_value val;
+    val.format = vpiIntVal;
+    vpiHandle rangeHdl;
```

Comment:
Not sure why these were moved up, this is less readable. Variable declarations should be as close to their use as possible.